version: "3.8"

services:
  desafioconta-service:
    build: .
    ports:
      - "8080:8080"
    depends_on:
      - mock-api
    networks:
      - banco-network
  mock-api:
    image: clue/json-server
    volumes:
      - ./mock-api/db.json:/data/db.json
      - ./mock-api/routes.json:/data/routes.json
    command: --watch /data/db.json --routes /data/routes.json --host 0.0.0.0
    ports:
      - 8081:80
    networks:
      - banco-network
  prometheus:
    image: prom/prometheus
    volumes:
      - ./prometheus/prometheus.yml:/etc/prometheus/prometheus.yml
    ports:
      - 9090:9090
    networks:
      - banco-network
  grafana:
    image: grafana/grafana
    ports:
      - "3001:3000"
    networks:
      - banco-network
  databasepostgres:
    image: postgres
    restart: always
    environment:
      POSTGRES_USER: user
      POSTGRES_HOST_AUTH_METHOD: trust
    networks:
      - banco-network
    ports:
      - "5432:5432"
    volumes:
      - "apimockbackend:/var/lib/postgresql/data"
      - "/var/run/postgres/postgres.sock:/var/run/postgres/postgres.sock"
networks:
  banco-network:
    driver: bridge

volumes:
  apimockbackend:
    driver: local
